nautobot/dcim/forms.py

```diff
@@ -140,6 +140,13 @@ def get_device_by_name_or_pk(name):
     return device
 
 
+class ExcludeIDMixin:
+    def __init__(self, exclude_id=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if exclude_id is not None:
+            self.fields["termination_b_id"].widget.add_query_param("id__n", str(exclude_id))
```

Comment:
If this is specific to cable connections (hard-coded `termination_b_id` suggests so) then let's name this something like `ConnectCableExcludeIDMixin` to be explicit. Alternatively, consider something like the following to make it more flexible:

```python
def __init__(self, *args, exclude_id=None, exclude_field="termination_b_id", **kwargs):
```